### DISCUSSION TITLE
How to disable an iOS App from being available as a Mac App? (Apple Silicon)

### DISCUSSION THREAD

**Discussion by squallstar** [fastlane/fastlane#16861] [Q&A] [answered]
Hello,

can you please let me know how I can edit the app availability through both Spaceship and Fastlane to ensure my iOS apps are not available as Mac apps?

This is following this new not from App Store Connect:

> Phone and iPad Apps on Mac. All newly created and existing compatible iPhone and iPad apps will be made available on Mac. Your compatible apps will be published on the Mac App Store on Apple Silicon Macs unless you edit their availability.

Thanks

**Comment by joshdholtz** [comment, marked as answer]
@squallstar Hey! This currently isn’t available through `Spaceship::ConnectAPI` but I will make a PR tomorrow morning to fix this 💪

**Reply by squallstar** [reply]
Thanks! Let me know once it's ready for testing.

**Reply by joshdholtz** [reply]
@squallstar 👉  https://github.com/fastlane/fastlane/pull/16973

**Reply by squallstar** [reply]
awesome! thanks

**Reply by nbaryshev** [reply]
Hey,
Sorry for being lame - I am new in fastlane.
Can someone share any examples of implementing this solution, please?

Hope this won't break any rules here. I have already written by another developer Fastfile. It contains different lanes. one of the is used to create new ios apps:

`lane :new_ios_app do |options|
  app_version = options[:app_version]
  config = get_app_config(options[:app_id])
  update_user_credentials(config)

  ios_meta = IosMetadata.new(config)
  ios_meta.deploy

  produce(
    app_name: config.app_name,
    username: ENV["USERNAME"],
    app_identifier: config.app_identifier,
    app_version: app_version,
    language: 'English',
    team_name: config.team_name,
    team_id: config.team_id,
    itc_team_name: config.itc_team_name,
    itc_team_id: config.itc_team_id,
    company_name: config.company_name,
    enable_services: {
      push_notification: "on",
      apple_pay: "off"
    }
  )

  deliver(
    force: true,
    username: ENV["USERNAME"],
    price_tier: 0,
    skip_binary_upload: false,
    app_version: app_version,
    app_icon: ios_meta.icon_path,
    metadata_path: ios_meta.metadata_path,
    app_identifier: config.app_identifier,
    app_rating_config_path: APP_RATING,
    team_name: config.team_name,
    team_id: config.itc_team_id
  )
end`

**Reply by joshdholtz** [reply]
@nbaryshev This is not added into either `produce` or `deliver` yet but you can do this manually with `Spaceship`. There is an example below 👇 

I should probably add this into `deliver` though 🤔 I’ll add that onto my queue!

```
require 'spaceship'

  Spaceship::ConnectAPI.login("your@email.com", team_name: “Your Team Name”)

  app = Spaceship::ConnectAPI::App.find("com.your.bundle.id")
  app.update(attributes: {
    is_opted_in_to_distribute_ios_app_on_mac_app_store: false
  })
```

**Reply by nbaryshev** [reply]
Hey @joshdholtz , it worked for a week and suddenly I started getting this error for each app I am trying to release. All these apps are based on one template we use in the company, so shouldn't be any changes.

The error is:
[14:37:26]: Failed to disable Mac -> #<Spaceship::UnexpectedResponse: The provided entity includes an unknown attribute - 'isOptedInToDistributeIosAppOnMacAppStore' is not an attribute on the resource 'apps'>

P.S let me know if I need to create a new question for this. I am a newbie and just didn't want to create a mess

Thanks!

**Reply by rogerluan** [reply]
@nbaryshev hey :) please do create a [new issue](https://github.com/fastlane/fastlane/issues/new?template=BUG_REPORT.md) so we can investigate this further. Thanks in advance!

**Reply by andrewdieken** [reply]
Hey, @joshdholtz! It would be great if this functionality was added to `produce` or `deliver`. Is there an issue or some way I'd be able to track the progress of this ask?

**Reply by Thompsons21** [reply]
Hello jennacoppolino,  [MonkeyCool](https://www.monkey.vin/)
Thanks for sharing such a helpful instruction, really appreciate for your. Exactly what I needed!
